src/geometry/instance.rs

```diff
@@ -0,0 +1,66 @@
+use std::rc::Rc;
+
+use geometry::mesh::Mesh;
+use geometry::{BoundingVolume, Intersectable, Shape, Transformable, TriangleStorage};
+use intersection::Intersection;
+use material::Material;
+use math::{Matrix4, Transform};
+use ray::Ray;
+
+pub struct Instance<V, S> {
+    mesh: Rc<Mesh<V, S>>,
+    world_to_view: Matrix4,
+    material: Rc<Material>,
+}
+
+impl<'a, V: BoundingVolume, S: TriangleStorage<'a>> Instance<V, S> {
+    pub fn new(mesh: Rc<Mesh<V, S>>, material: Rc<Material>) -> Self {
+        Self {
+            mesh,
+            world_to_view: Matrix4::identity(),
+            material,
+        }
+    }
+}
+
+impl<V: BoundingVolume, S: for<'a> TriangleStorage<'a>> Transformable for Instance<V, S> {
```

Comment:
Why did you use `S: for<'a> TriangleStorage<'a>` instead of `'a, S: TriangleStorage<'a>` or `S: TriangleStorage<'_>`?

Reply:
tbh I don't remember but getting the code around the TriangleStorage lifetimes to work correctly was complex and this works(in this instance it's copied from `src/geometry/mesh.rs`)